imap/jmap_mailbox.c

```diff
@@ -152,9 +152,17 @@ HIDDEN void jmap_mailbox_init(jmap_settings_t *settings)
     }
 }
 
-HIDDEN void jmap_mailbox_capabilities(json_t *account_capabilities
-                                      __attribute__((unused)))
+HIDDEN void jmap_mailbox_capabilities(json_t *account_capabilities,
+                                      int mayCreateTopLevel)
 {
+    json_t *email_capa =
+        json_object_get(account_capabilities, JMAP_URN_MAIL);
+
+    json_object_set_new(email_capa, "maxMailboxDepth", json_null());
+    json_object_set_new(email_capa, "maxSizeMailboxName",
+                        json_integer(MAX_MAILBOX_NAME / 2));
```

Comment:
Maybe it's too early morning for me, but why do we only allow half of MAX_MAILBOX_NAME?

Reply:
So MAX_MAILBOX_NAME is the limit on the entire mailbox name (e.g. "user.userid.foo.bar"), not just each individual name segment (e.g. "foo").  Since JMAP never exposes "user.userid", we have to set our limit less than MAX_MAILBOX_NAME.  And Rik did some research and founf that we have at least one user with a mailhbox name segment of 247 chars.  So we chose MAX_MAILBOX_NAME/2 to accommodate this nonsense.

Reply:
I figured it must be something like that. WFM.